core/class/scenario.class.php

```diff
@@ -840,6 +840,7 @@ public function launch($_trigger = '', $_message = '', $_forceSyncMode = false,
 			$this->setTags($tags);
 		}
 		$this->setCache(array('startingTime' => strtotime('now'), 'state' => 'starting'));
+		$this->setCache('lastExecutionUser', (isset($_SESSION) && isset($_SESSION['user']) && $_SESSION['user'] != null) ? $_SESSION['user']->getLogin() : 'none');
```

Comment:
Consider handling the case where `$_SESSION['user']` is not set more gracefully. Instead of setting the `lastExecutionUser` to 'none', it might be better to leave it unset or set it to a more meaningful value. [medium]